scene/gui/text_edit.cpp

```diff
@@ -1689,6 +1725,19 @@ void TextEdit::gui_input(const Ref<InputEvent> &p_gui_input) {
 			hovered_gutter = current_hovered_gutter;
 			update();
 		}
+
+		if (drag_action && can_drop_data(mpos, get_viewport()->gui_get_drag_data())) {
+			drag_caret_force_displayed = true;
+			Point2i pos = get_line_column_at_pos(get_local_mouse_pos());
+			set_caret_line(pos.y, false);
+			set_caret_column(pos.x);
+			if (pos.y <= get_first_visible_line()) {
+				_scroll_lines_up();
+			} else if (pos.y >= get_last_full_visible_line()) {
+				_scroll_lines_down();
+			}
```

Comment:
We shouldn't need to do this, `set_caret_column` should adjust the viewport for us.

Reply:
Yes, in theory it shouldn't be necessary, but in practice the scroll won't work without this code.
I have the possibility to test only on Linux, so I don't know if the problem is limited to this platform, but without the code I added the scroll only works if you move the selection from top to bottom, if you move from the bottom upward the scroll does not work.
I can add a comment at the beginning of the code to indicate that this is a temporary workaround, if you remove that code the drag functionality would be almost unusable.

Reply:
I would still remove it for this PR, not a keen on adding temporary workarounds as they quickly become permanent. The base issue should be fixed instead.

Reply:
done